nservicebus/architecture/publishing-from-web-applications.md

```diff
@@ -0,0 +1,89 @@
+
+But, things change:
+
+* With IIS 7.5 and later, IIS supports AlwaysRunning mode.
+* IIS is no longer the only way to host web applications, which can now be hosted by [OWIN](http://owin.org/)/[Katana](http://www.asp.net/aspnet/overview/owin-and-katana), [NancyFx](http://nancyfx.org/), [FubuMVC](http://mvc.fubu-project.org/), and others. This largely removes the concern over web application recycling.
+* NServiceBus 4.0 introduced the ability to have transports besides MSMQ. Not all of the transports handle Publish/Subscribe in the same way as MSMQ - some even support it natively.
+* Cloud transports generally have an associated cost per interaction. This means that requiring an unnecessary send before publish costs extra money to operate.
+* In NServiceBus 4.0 and 5.0, the process of creating a send-only endpoint (which is still capable of publishing messages, but does not have an input queue or process incoming messages) was made much easier.
+
+## Publish/Subscribe Mechanics
+
+First it's important to understand how publish/subscribe actually works, consisting of the subscription phase, and then the actual publish operation.
+
+### Subscription
+
+An endpoint must register its interest in a message by subscribing to a specific message type. How this occurs is dependent upon the message transport being used.
+
+Some transports (Azure Service Bus, RabbitMQ) support publish/subscribe natively. This means that when an endpoint wants to subcribe to an event, it contacts the broker directly, and the broker maintains the list of subscriptions.
+
+Other transports (MSMQ, SQL, Azure Storage Queues) do not support native publish/subscribe, and instead use storage-driven publishing with message-driven subscriptions. This means that each endpoint is responsible for maintaining its own subscription storage, usually in a database. When an endpoint wants to subscribe to an event, it sends a subscription request message to the owner endpoint, which will update its own subscription storage.
+
+### Publishing
+
+The publishing process also differs based on the message transport.
+
+A transport that supports native publish/subscribe (Azure Service Bus, RabbitMQ) will send the event message directly to the broker, addressed to a topic, which will distribute the message to all interested parties.
+
+A transport that uses storage-driven publishing will act differently. When it needs to publish, it will first consult its subscription storage to determine the interested parties. It will then send an independent copy of the event message to each subscriber.
+
+While it is common to think of an event being published *from* a specific location, this doesn't always reflect the reality of the messaging mechanics. For native pub/sub brokers, the "publish from" location (the `Endpoint` value in the `MessageEndpointMappings` section) relates to a Topic in the broker.

```

Comment:
Not correct, you need no mappings at all for Pub/Sub to work with the brokers since you just need to know the event type. The broker then figures out the rest